aten/src/ATen/native/TensorShape.cpp

```diff
@@ -346,6 +344,74 @@ Tensor cat(TensorList tensors, int64_t dim) {
   return result;
 }
 
+Tensor block_diag(TensorList tensors) {
+  Tensor result = at::empty({}, tensors[0].options());
+  return at::block_diag_out(result, tensors);
+}
+
+Tensor& block_diag_out(Tensor& result, TensorList tensors) {
+  if (tensors.size() == 0) {
+    result.resize_({0});
+    return result;
+  }
+
+  int64_t result_dim0 = 0;
+  int64_t result_dim1 = 0;
+  ScalarType scalar_type = tensors[0].scalar_type();
+  std::vector<Tensor> tensors_2D(tensors.size());
+
+  // Sum the dimensions of the tensors, check tensor sizes,
+  // and expand all 0-D and 1-D tensors so that everything
+  // is 2-D
+  for (size_t tensor_idx = 0; tensor_idx < tensors.size(); tensor_idx++) {
+    Tensor tensor = tensors[tensor_idx];
+    int64_t ndims = tensor.dim();
+    TORCH_CHECK(scalar_type == tensor.scalar_type(), "tensors must all have the same scalar type");
+    TORCH_CHECK(ndims <= 2, "tensors must have 2 or fewer dimensions");
+
+    int64_t dim0 = 1;
+    int64_t dim1 = 1;
+
+    if (ndims == 2) {
+      dim0 = tensor.size(0);
+      dim1 = tensor.size(1);
+
+      TORCH_CHECK(dim0 > 0, "2-D must have size(0) > 0")
```

Comment:
Hidden dangers of inconsistent behavior:
```
from scipy.linalg import block_diag
A = []
B = [1, 0]
block_diag(A,B)
>>> array([[0., 0.], [1., 0.]])
```
Scipy allows empty arrays as input or even output. Personally, I recommend the implementation of scipy, because the subgraph is likely to be empty during preprocessing

Reply:
Thanks for pointing that out, I'll change it.